lib/datasource/metadata.ts

```diff
@@ -117,6 +118,43 @@ function massageGitlabUrl(url: string): string {
     .replace('.git', '');
 }
 
+function normalizeDate(input: any): string | null {
+  if (
+    typeof input === 'number' &&
+    !Number.isNaN(input) &&
+    input > 0 &&
+    input <= Date.now() + 24 * 60 * 60 * 1000
+  ) {
+    return new Date(input).toISOString();
+  }
+
+  if (typeof input === 'string') {
+    const luxonDate = DateTime.fromISO(input, { zone: 'UTC' });
+    if (luxonDate.isValid) {
+      return luxonDate.toISO();
+    }
+
+    return normalizeDate(Date.parse(input));
```

Comment:
What's the advantage of this "recursion"?

Reply:
The advantage is just to make logic of the function more "flat". Though, if coercing from `number` (i.e. Unix epoch) is undesirable, I'll get rid of that.